File: MFDLabs.Grid.Bot/Commands/CloseAllJobs.cs
using System.Threading.Tasks;
using Discord.WebSocket;
using MFDLabs.Grid.Bot.Extensions;
using MFDLabs.Grid.Bot.Interfaces;
using MFDLabs.Grid.Bot.Utility;

namespace MFDLabs.Grid.Bot.Commands
{
    internal sealed class CloseAllJobs : IStateSpecificCommandHandler
    {
        public string CommandName => "Close All Grid Server Jobs";
        public string CommandDescription => "Attempts to close all of the grid server jobs via the SoapUtility.";
        public string[] CommandAliases => new[] { "caj", "closealljobs" };
        public bool Internal => true;
        public bool IsEnabled { get; set; } = true;

        public async Task Invoke(string[] messageContentArray, SocketMessage message, string originalCommand)
        {
            if (!await message.RejectIfNotAdminAsync()) return;

            await GridServerArbiter.Singleton.CloseAllJobsAsync();
            await message.ReplyAsync("Successfully closed all grid server jobs.");
        }
    }
}

File: Source/Lib/Commands/CloseAllJobs.ts
import { Message } from 'discord.js';
import { AdminUtil } from '../Administration/AdminUtil';
import { SOAPUtil } from '../RCCService/SOAPUtility';
import { StateSpecificCommandHandler } from '../CommandUtility/Types/StateSpecificCommandHandler';

class CloseAllJobs implements StateSpecificCommandHandler {
	public CommandName = 'Close All Jobs';
	public CommandDescription = 'Closes all jobs';
	public Internal = true;
	public Command = ['caj', 'closealljobs', 'closeallgames'];
	public Callback = async (messageContent: string[], messageRoot: Message) => {
		if (!AdminUtil.RejectIfNotAdmin(messageRoot)) return;

		await SOAPUtil.CloseAllJobs();
		return messageRoot.reply(`Successfully closed all jobs.`);
	};
}

export = new CloseAllJobs();
